### PR TITLE
rustdoc: use slice::contains instead of open-coding it

### PR DESCRIPTION
**Pull request by notriddle**

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

r=me once CI pass.

File: src/librustdoc/passes/html_tags.rs [hunk 1]
```diff
@@ -207,7 +208,7 @@ impl<'a, 'tcx> DocFolder for InvalidHtmlTagsLinter<'a, 'tcx> {
 
             for (tag, range) in tags.iter().filter(|(t, _)| {
                 let t = t.to_lowercase();
-                ALLOWED_UNCLOSED.iter().find(|&&at| at == t).is_none()
+                !ALLOWED_UNCLOSED.iter().contains(&t.as_str())
```
Comment (cuviper) [inline src/librustdoc/passes/html_tags.rs:211]: Better yet, just use the slice [`contains`](https://doc.rust-lang.org/std/primitive.slice.html#method.contains).

There's another instance with `ALLOWED_UNCLOSED` like this in `fn drop_tag`.

**Comment by notriddle** [comment]
Sorry about that. I probably should've noticed that `slice::contains` existed.

**Pull request merged**